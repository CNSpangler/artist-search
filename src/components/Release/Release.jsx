import React from 'react';
import PropTypes from 'prop-types';

const Release = ({ songs, releaseId }) => {
  const songElements = songs.map(song => (
    <li key={song.songId}>
      {song.songTitle}
    </li>
  ));

  return (
    <>
      <img src={`http://coverartarchive.org/release/${releaseId}/front`} />
      <ul>
        {songElements}
      </ul>
    </>
  );
};

Release.propTypes = {
  songs: PropTypes.array.isRequired,
  releaseId: PropTypes.string.isRequired,
};

export default Release;
