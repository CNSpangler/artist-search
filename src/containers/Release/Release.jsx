import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';
import { fetchSongs } from '../../services/APIcalls';
import TrackList from '../TrackList/TrackList.jsx';

const Release = () => {
  const [tracks, setTracks] = useState([]);
  const { releaseId } = useParams();

  useEffect(() => {
    fetchSongs(releaseId)
      .then(results => setTracks(results));
  }, []);

  return <TrackList tracks={tracks} />;
};

Release.propTypes = {
  songs: PropTypes.arrayOf(PropTypes.shape({
    songId: PropTypes.string.isRequired,
    songTitle: PropTypes.string.isRequired
  })).isRequired,
  releaseId: PropTypes.string.isRequired,
};

export default Release;
