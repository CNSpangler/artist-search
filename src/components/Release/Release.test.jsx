import React from 'react';
import { shallow } from 'enzyme';
import Release from './Release.jsx';

describe('Release component', () => {
  it('renders Release', () => {
    const wrapper = shallow(<Release songs={[
      {
        songId: 1234,
        songTitle: 'Help'
      },
      {
        songId: 1235,
        songTitle: 'Halp'
      }
    ]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
