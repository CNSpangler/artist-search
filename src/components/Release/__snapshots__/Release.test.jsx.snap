// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component renders Release 1`] = `
<Fragment>
  <img
    src="http://coverartarchive.org/release/undefined/front"
  />
  <ul>
    <li
      key="1234"
    >
      Help
    </li>
    <li
      key="1235"
    >
      Halp
    </li>
  </ul>
</Fragment>
`;
